hera_cal/reflections.py

```diff
@@ -398,37 +749,76 @@ def fit_reflection(data, dly_range, freqs, full_freqs=None, edgecut=0,
     # https://ccrma.stanford.edu/~jos/sasp/Quadratic_Interpolation_Spectral_Peaks.html
     # alpha = a, beta = b, gamma = g
     a = abs_dfft[np.arange(Ntimes), ref_dly_inds - 1, None]
-    g = abs_dfft[np.arange(Ntimes), ref_dly_inds + 1, None]
+    g = abs_dfft[np.arange(Ntimes), (ref_dly_inds + 1) % Ndlys, None]
     b = abs_dfft[np.arange(Ntimes), ref_dly_inds, None]
-    bin_shifts = 0.5 * (a - g) / (a - 2 * b + g)
+    denom = (a - 2 * b + g)
+    bin_shifts = 0.5 * np.true_divide((a - g), denom, where=~np.isclose(denom, 0.0))
 
     # update delay center and peak value given shifts
     ref_dlys += bin_shifts * np.median(np.diff(dlys))
     ref_peaks = b - 0.25 * (a - g) * bin_shifts
 
-    # get reflection significance, defined as ref_peak / median_abs
-    ref_significance = ref_peaks / np.median(abs_dfft_selection, axis=1, keepdims=True)
+    if return_peak:
+        return ref_peaks, ref_dlys
+
+    # get reflection significance, defined as max|V| / med|V|
+    avgmed = np.median(abs_dfft_selection, axis=1, keepdims=True)
+    ref_significance = np.true_divide(ref_peaks, avgmed, where=~np.isclose(avgmed, 0.0))
+
+    # get peak value at tau near zero
+    peak, _ = fit_reflection_delay(dfft, (-np.abs(dly_range).min(), np.abs(dly_range).min()), dlys, return_peak=True)
 
     # get reflection amplitude
-    ref_amps = ref_peaks / np.max(abs_dfft[:, np.abs(dlys) < np.abs(dly_range).min()], axis=1, keepdims=True)
+    ref_amps = np.true_divide(ref_peaks, peak, where=~np.isclose(peak, 0.0))
+
+    return ref_amps, ref_dlys, ref_dly_inds, ref_significance
+
+
+def fit_reflection_phase(dfft, dly_range, dlys, ref_amps, ref_dlys, fthin=1, Nphs=500,
```

Comment:
This whole algorithm feels a bit janky. What kind of accuracy can you actually get? Presumably no better than 2pi over the number of phase bins...

I'd recommend trying `scipy.optimize.minimize` or `scipy.optimize.curve_fit` or similar. Should be fairly straightforward to fit a cosine rather than generating a grid of cosines.

Reply:
So the adopted algorithm is done for speed: this is the fastest thing I could get that hits the necessary precision, I experimented with `scipy` minimization algorithms at one point and found it significantly slower. Precision is indeed 2pi / `Nphs`, which is exposed to the user in case higher precision is required...

Reply:
I've done tests and the phase is actually not the limiting factor for sufficiently high `Nphs` (the default parameter). But sure, the quadratic interpolation can't hurt..